### ISSUE TITLE
client.write() does not work with python 2.7

### ISSUE THREAD

**Issue by Notargets** [alpacahq/pymarketstore#20]
When I try the example in the README, it fails with a corrupted time index:

```
import numpy as np
import pandas as pd
import pymarketstore as pymkts

data = np.array([(pd.Timestamp('2017-01-01 00:00').value / 10**9, 10.0)], dtype=[('Epoch', 'i8'), ('Ask', 'f4')])

cli = pymkts.Client()

cli.write(data, 'TEST/1Min/Tick')

d2 = cli.query(pymkts.Params('TEST', '1Min', 'Tick')).first().df()
print d2
```

I get the following output:
```
# python tests/basic_driver.py 
Empty DataFrame
Columns: [Ask]
Index: []
```

I believe the problem is in the handling of the numpy data in the driver as compared with the handling in the API code in the marketstore server. To fix this, we should:
- develop and document the binary data interchange format
- implement a basic sanity test of the index values used in the data package - perhaps send the beginning and ending values of time index in the intended write data in the data package
- implement a system level test in CircleCI that verifies this driver's correct operation with the latest marketstore codebase

**Comment by umitanuki** [comment]
Hmm, I cannot reproduce it in my env.

```
[pymarketstore: hitoshi{master}]$ python basic_driver.py
                            Ask
Epoch
2017-01-01 00:00:00+00:00  10.0
```

This is what my env looks like.

```
[pymarketstore: hitoshi{master}]$ pip list
Package          Version
---------------- ---------
certifi          2018.8.24
chardet          3.0.4
idna             2.7
msgpack-python   0.5.6
numpy            1.15.1
pandas           0.23.4
pip              18.0
python-dateutil  2.7.3
pytz             2018.5
requests         2.19.1
setuptools       40.2.0
six              1.11.0
urllib3          1.23
websocket-client 0.53.0
wheel            0.31.1

[pymarketstore: hitoshi{master}]$ python -V
Python 3.6.6
```

**Comment by Notargets** [comment]
Yep - seems like it might depend on versions of things - here's my env:
```
$ pip list | grep -E 'msgpack-python|numpy|pandas|python-dateutil|pytz|websocket-client'
msgpack-python                     0.5.6
numpy                              1.15.1
pandas                             0.23.4
python-dateutil                    2.7.3
pytz                               2018.5
websocket-client                   0.52.0

$ python -V
Python 2.7.14
```

**Comment by umitanuki** [comment]
Thanks.

Here is the different for `writer` in 2 and 3.

## python2
```
In [2]: writer
Out[2]:
{'requests': [{'dataset': {'data': ['<memory at 0x7f1d9ae000e8>',
     '<memory at 0x7f1d9ae000e8>'],
    'length': 1,
    'lengths': {'TEST/1Min/Tick': 1},
    'names': ('Epoch', 'Ask'),
    'startindex': {'TEST/1Min/Tick': 0},
    'types': ['i8', 'f4']},
   'isvariablelength': False}]}
```

## python3
```
In [1]: writer
Out[1]:
{'requests': [{'dataset': {'types': ['i8', 'f4'],
    'names': ('Epoch', 'Ask'),
    'data': [b'\x80FhX\x00\x00\x00\x00', b'\x00\x00 A'],
    'length': 1,
    'startindex': {'TEST/1Min/Tick': 0},
    'lengths': {'TEST/1Min/Tick': 1}},
   'isvariablelength': False}]}
```

**Comment by umitanuki** [comment]
In python2, there is no byte array, and it looks like the correct way to get the pseudo byte array in str type is

```
In [7]: bytes(buffer(recarray['Epoch']))
Out[7]: '\x80FhX\x00\x00\x00\x00'

In [8]: bytes(buffer(recarray['Ask']))
Out[8]: '\x00\x00 A'
```

**Comment by umitanuki** [comment]
Fixed in #21
